homeworks/Aleksandr.Turuta_turuta7/homework_5/home_5.js

```diff
@@ -0,0 +1,75 @@
+// the task 1
+function createCounter() {
+    let num = 0;
+    return ((ppp) => {
+        num += ppp;
+        return num;
+    });
+}
+
+function sumArray(array) {
+    const res = array;
+    const fn = createCounter();
+    for (let i = 0; i < res.length; i++) {
+        res[i] = fn(res[i]);
+    }
+    return res;
+}
+
+
+console.log(sumArray([10, -10, 10, -10, 10]));
+console.log(sumArray([0, 0, 0, 0, 0]));
+console.log(sumArray([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]));
+
+// the task 2
+const employee = ['header', 'menu', 'menu_item', 'tabs',
+    'tab_item', 'menu', 'link', 'tabs', 'tab_item', 'menu',
+    'menu_item', 'menu', 'menu_item'];
+
+function arr(ob) {
+    const res = {};
+    ob.forEach((x) => {
+        if (x in res) {
+            res[x] += 1;
+        } else {
+            res[x] = 1;
+        }
+    });
+    return res;
+}
+console.log(arr(employee));
+
+// the task 3
+const Array = { b: 'c', d: { e: 'f' } };
+
+function newArray(arr1) {
+    const newObject = {};
+    Object.keys(arr1).forEach((x) => {
+        newObject[x] = arr1[x];
+    });
+    return newObject;
+}
+console.log(newArray(Array));
```

Comment:
deep copying of objects is not that simple, please read about variables as references to objects